connectors/kafka/src/main/java/forklift/message/KafkaMessage.java

```diff
@@ -0,0 +1,67 @@
+package forklift.message;
+
+import forklift.connectors.ConnectorException;
+import forklift.connectors.ForkliftMessage;
+import forklift.consumer.parser.KeyValueParser;
+import forklift.controller.KafkaController;
+import forklift.producers.KafkaForkliftProducer;
+import org.apache.avro.generic.GenericRecord;
+import org.apache.kafka.clients.consumer.ConsumerRecord;
+
+public class KafkaMessage extends ForkliftMessage {
+    private final KafkaController controller;
+    private final ConsumerRecord<?, ?> consumerRecord;
+
+    public KafkaMessage(KafkaController controller, ConsumerRecord<?, ?> consumerRecord) {
+        this.controller = controller;
+        this.consumerRecord = consumerRecord;
+        parseRecord();
+    }
+
+    public ConsumerRecord<?, ?> getConsumerRecord() {
+        return this.consumerRecord;
+    }
+
+    @Override
+    public boolean acknowledge() throws ConnectorException {
+        try {
+            return controller.acknowledge(consumerRecord);
+        } catch (InterruptedException e) {
+            throw new ConnectorException("Error acknowledging message");
+        }
+    }
+
+    @Override
+    public String getId() {
+        return consumerRecord.topic() + "-" + consumerRecord.partition() + "-" + consumerRecord.offset();
+    }
+
+    /**
+     * <strong>WARNING:</strong> Called from constructor
```

Comment:
Mainly as a warning to other developer.  I suppose marking it final serves as enough warning...I just in general dislike calling methods from the constructor